name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/wizards/datatransfer/ReviewExtractedItemsPage.java

```diff
@@ -496,17 +538,42 @@ public void beforePage()
                 protected IStatus run(IProgressMonitor monitor)
                 {
                     monitor.beginTask(Messages.PDFImportWizardMsgExtracting, files.size());
-
                     final List<Exception> errors = new ArrayList<Exception>();
-                    List<ExtractedEntry> entries = extractor //
-                                    .extract(files, errors).stream() //
-                                    .map(i -> new ExtractedEntry(i)) //
-                                    .collect(Collectors.toList());
-
-                    // Logging them is not a bad idea if the whole method fails
-                    PortfolioPlugin.log(errors);
-
-                    Display.getDefault().asyncExec(() -> setResults(entries, errors));
+                    
+                    try
+                    {
+                        if (extractor.getLabel() == "pdfimportassistant")
+                        {
+                            new PDFImportAssistant();
+                            Extractor extractorDropDown = PDFImportAssistant.createExtractor(SelectedBankExtractor, client);
+                            List<ExtractedEntry> entries = extractorDropDown //
+                                            .extract(files, errors).stream() //
```

Comment:
Wenn ich das richtig sehe, dann wird das PDF jetzt zweimal in Text umgewandelt. Einmal um den Extractor zu bestimmen - und dann noch mal im Extractor selber. Das ist eine relativ teure Operation - ich denke das sollte man eher nur einmal machen.

Allerdings gibt es ja mit pdfbox 2.0.7 ggf. die Komplikation von falschen Unicode maps. Siehe auch hier: https://github.com/buchen/portfolio/pull/807#issuecomment-326724141 Das hat Auswirkungen auf die Bank Identifier und ggf. muss der Extractor das Original PDF noch mal ändern.

Vielleicht reicht man ein Object an den Extractor, dass beides beinhaltet: File und Text?